scripts/device-common-install/libexec/device_hw_validation.sh

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bash
+
+is_expected_hdd() {
+  local expected_value
+  local tolerance
+
+  expected_value="${1?ERROR variables is unset}"
```

Comment:
Note that as this is more of an assertion (programmer error) than an actual possible error, you can also leave the message empty which should yield a similar stderr:

```bash
$ my_fn 
bash: 1: parameter null or not set
```